docs/concepts/content-addressing.md

```diff
@@ -69,66 +69,73 @@ $ ipfs cid format -v 1 -b base32 QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR
+:::callout
+**See CID conversion in action**
+See the [interactive code sandbox](#codesandbox-converting-between-cid-versions-and-encodings) for an example JS application that converts between CID versions and encodings.
+:::
 
-To convert a CIDv1 from text to binary form, simply read the first character
-and then decode the remainder using the encoding specified in the [multibase table](https://github.com/multiformats/multibase#multibase-table).
+### Converting between CID base encodings
 
-JS users can leverage the [`cids`](https://www.npmjs.com/package/cids) library to get a binary version as `Uint8Array`:
+A CID can be encoded using any of the encodings specified in the [multibase table](https://github.com/multiformats/multibase#multibase-table). The use of different encodings can impact speed and storage efficiency.
 
+To convert a CIDv1 `cidV1` from one encoding to another, use the `toString()` method. By default, `toString()` will return the `base32` string representation of the CID, but you can use other string representations:
 
 ```js
-const CID = require('cids')
-new CID('bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi').bytes
-// → Uint8Array [ 1, 112,  18,  32, 195, 196, 115,  62, ... ]
+const cidV1StringBase32 = cidV1.toString();
 ```
 
-::: warning Be mindful about parsing CIDs correctly. Avoid shortcuts.
-
-Unless you are the one who imported the data to IPFS, the length of a CID is not deterministic and depends on the length of the multihash inside of it.
+The following example returns the base256 emoji encoding of the CID:
+```js
+const cidV1StringBase256 = cidV1.toString(base256emoji);
+```
 
-To illustrate, passing a custom hash function will produce CIDs of varying lengths:
+Using `.bytes`, the following example returns the raw bytes of the CID:
 
+```js
+const cidV1Bytes = cidV1.bytes
 ```
-$ ipfs add --cid-version 1 --hash sha2-256    -nq cat.jpg | wc -c
-60
-$ ipfs add --cid-version 1 --hash blake2b-256 -nq cat.jpg | wc -c
-63
-$ ipfs add --cid-version 1 --hash sha3-512    -nq cat.jpg | wc -c
-111
-```
+
+:::callout
+**See CID conversion in action**
+See the [interactive code sandbox](#codesandbox-converting-between-cid-versions-and-encodings) for an example JS application that converts between CID versions and encodings.
 :::
 
 
 ### CID to hex
 
 Sometimes, a [hexadecimal](https://en.wikipedia.org/wiki/Hexadecimal) representation of raw bytes is preferred for debug purposes.
-To get the hex for raw `.bytes` of an entire CID, one can use built-in support for `base16` encoding and skip the `f` (multibase prefix):
-
-```javascript
-> cid.toString('base16')
-'f01701220c3c4733ec8affd06cf9e9ff50ffc6bcd2ec85a6170004bb709669c31de94391a'
+To get the hex for raw `.bytes` of a CIDv1 `cidV1`, use `base16` encoding:
 
-> cid.toString('base16').substring(1)
-'01701220c3c4733ec8affd06cf9e9ff50ffc6bcd2ec85a6170004bb709669c31de94391a' // "cid as hex"
+```js
+const cidV1StringBase256 = cidV1.toString(base16);
```

Comment:
I think it may make sense to show where `base16` comes from (the import) and maybe even the how cidV1 is instantiated?